packages/application-shell-connectors/src/components/application-state/application-state.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { wrapDisplayName } from 'recompose';
+import moment from 'moment-timezone';
+import omit from 'lodash.omit';
+
+const { Provider, Consumer } = React.createContext({});
+
+const defaultTimeZone = moment.tz.guess() || 'Etc/UTC';
+
+// Expose only certain fields as some of them are only meant to
+// be used internally in the AppShell
+const mapUserPropsToApplicationState = user => ({
+  id: user.id,
+  email: user.email,
+  firstName: user.firstName,
+  lastName: user.lastName,
+  // NOTE: this is an alias for the original field `user.language`.
+  // This field is used for the application intl locale though, therefore
+  // we might want to expose it with a more explicit name to avoid
+  // confusion.
+  applicationLocale: user.language,
```

Comment:
I feel like exposing a more explicit field would make it easier to understand what it does.

Reply:
I'd like to hear @dferber90's thoughts cause he's put some 💭 into the naming of languages across the app.

Reply:
Alright, I'm fine with `user.locale`. I though that naming it `applicationLocale` would make it more explicit.